### PR TITLE
Fix focus group for Notebook in LMS context

### PR DESCRIPTION
**Pull request by LMS007**
Now that the config passes the `group` value from the LMS iframe, we can use it to set the initial focus group. This still was not flushed out and I traced it down to how we load the groups service. 

### Testing

**This particular issue is very specific to set up.**
1. Requires canvas (or any LMS that can launch the assignment in the same window as the LMS)
2. Requires third party cookies to be disabled (to prevent localStorage from working in iframes with different hostnames)
3. **Do not** load the canvas assignment in a new window  because then localStorage will work and you can't repo this exact situation 

I use this assignment: https://hypothesis.instructure.com/courses/125/assignments/874

Select any group except for the first, then open the notebook and observe that that group is indeed used in the notebook and so are the annotations.

Secondly, observe that the notebook still works in http://localhost:3000 for non LMS usage. 

----------

### Commit

Use directLinkedGroupId when loading groups

Change loadServiceSpecifiedGroups() so that it can optionally pass along a group ID setting value from the `directLinkedGroupId` selector to be an initially focused.

When loading groups from a service e.g. `loadServiceSpecifiedGroups`, previously it assumed that the `directLinkedGroupId` value would not be used in this context. Since the introduction of the Notebook and specifically in the context of LMS, we have both a service setting which contains the groups  (`settings.services[0].groups <Promise>`)
and a top level group to set focus too. (`settings.group`).

fixes https://github.com/hypothesis/client/issues/3251

### CODE REVIEW

**Review by robertknight — APPROVED** [review]
The code changes look good to me and I understand the issue from looking at them and comparing the use of `addGroupsToStore` in different places in the service. I found the explanation of the issue hard to parse from the first paragraph of the PR description though.

I haven't re-tested this manually myself but I trust that you have!

File: src/sidebar/services/groups.js [hunk 1]
```diff
@@ -366,7 +366,10 @@ export default function groups(
       g => g !== null
     );
 
-    addGroupsToStore(groups);
+    // Optional direct linked group id. This is passed in the Notebook context.
```
Comment (lyzadanger) [inline src/sidebar/services/groups.js:369]: I'm left feeling unsure what "passed" means in this comment.

Comment (LMS007) [reply src/sidebar/services/groups.js:369]: perhaps "used" would be better
